src/Test/Utilities/Portable/Compilation/ControlFlowGraphVerifier.cs

```diff
@@ -20,20 +22,51 @@ public static void VerifyGraph(Compilation compilation, string expectedFlowGraph
         public static string GetFlowGraph(Compilation compilation, ControlFlowGraph graph)
```

Comment:
The abundance of local functions interspersed with logic, with further nested local functions and deeply nested conditional logic smells, in my opinion.
Can't the code be factored in some more sensible way?

Reply:
> Can't the code be factored in some more sensible way?

Do you have any specific suggestion in mind? 

---
In reply to: [167717665](https://github.com/dotnet/roslyn/pull/24770#discussion_r167717665) [](ancestors = 167717665)

Reply:
For starters, I'd avoid nested local functions. 
On first examination, `buildRegionMap`, `enterRegions` and `leaveRegions` can be top-level static private method methods. Probably `validateBranch` and `validateRoot` too.
Create a visitor if needed.
If local functions remain, move them at the bottom of the method (or top is fine too).

Reply:
Thanks, this made the code much easier to follow (for me at least).

---
In reply to: [167720289](https://github.com/dotnet/roslyn/pull/24770#discussion_r167720289) [](ancestors = 167720289)